compiler/rustc_mir/src/const_eval/machine.rs

```diff
@@ -309,6 +309,35 @@ impl<'mir, 'tcx> interpret::Machine<'mir, 'tcx> for CompileTimeInterpreter<'mir,
                 );
                 ecx.write_scalar(Scalar::Ptr(ptr), dest)?;
             }
+            sym::panic_ctfe_hook => {
+                // Option<&str>
+                assert!(args.len() == 1);
+
+                debug!("panic_ctfe_hook: invoked with {:?}", args[0]);
+
+                let (_, idx) = ecx.read_discriminant(&args[0])?;
+                let some = match idx.index() {
+                    0 => {
+                        // None
+                        // For now, this is unreachable code - you cannot construct a non-literal
```

Comment:
Why is this unreachable? What happens when I do `panic!("{}", "hello")` in a `const fn`?